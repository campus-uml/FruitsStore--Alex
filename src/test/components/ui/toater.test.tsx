import { render, screen } from "@testing-library/react"
import { describe, it, vi, Mock, expect } from "vitest"
import "@testing-library/jest-dom"
import { Toaster } from "../../../components/ui/toaster"
import { useToast } from "@/hooks/use-toast"

vi.mock("@/hooks/use-toast")

describe("Toaster component", () => {
    it("renders without crashing", () => {
        (useToast as Mock).mockReturnValue({ toasts: [] })
        render(<Toaster />)
        const region = screen.getByRole("region");
        expect(region).toBeInTheDocument();
        expect(region).toMatchSnapshot();

    })

    it("renders toast messages", () => {
        (useToast as Mock).mockReturnValue({
            toasts: [
                { id: 1, title: "Toast 1", description: "Description 1" },
                { id: 2, title: "Toast 2", description: "Description 2" },
            ],
        })
        render(<Toaster />)
        expect(screen.getByText("Toast 1")).toBeInTheDocument()
        expect(screen.getByText("Description 1")).toBeInTheDocument()
        expect(screen.getByText("Toast 2")).toBeInTheDocument()
        expect(screen.getByText("Description 2")).toBeInTheDocument()
        expect(screen.getByText("Description 2")).toMatchSnapshot()

    })

    it("renders toast with action", () => {
        (useToast as Mock).mockReturnValue({
            toasts: [
                {
                    id: 1,
                    title: "Toast with Action",
                    description: "Description with Action",
                    action: <button>Action</button>,
                },
            ],
        })
        render(<Toaster />)
        expect(screen.getByText("Toast with Action")).toBeInTheDocument()
        expect(screen.getByText("Description with Action")).toBeInTheDocument()
        expect(screen.getByRole("button", { name: /action/i })).toBeInTheDocument()
        expect(screen.getByRole("button", { name: /action/i })).toMatchSnapshot()
    })
})
