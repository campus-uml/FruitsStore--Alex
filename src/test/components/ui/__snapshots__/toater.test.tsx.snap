// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toaster component > renders toast messages 1`] = `
<div
  class="text-sm opacity-90"
>
  Description 2
</div>
`;

exports[`Toaster component > renders toast with action 1`] = `
<button>
  Action
</button>
`;

exports[`Toaster component > renders without crashing 1`] = `
<div
  aria-label="Notifications (F8)"
  role="region"
  style="pointer-events: none;"
  tabindex="-1"
>
  <ol
    class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
    tabindex="-1"
  />
</div>
`;
